pkg/common/cluster/healthchecks/healthcheckjob.go

```diff
@@ -0,0 +1,67 @@
+package healthchecks
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/openshift/osde2e/pkg/common/logging"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+)
+
+// CheckHealthcheckJob uses the `openshift-cluster-ready-*` healthcheck job to determine cluster readiness.
+func CheckHealthcheckJob(k8sClient *kubernetes.Clientset, ctx context.Context, logger *log.Logger) (bool, error) {
+	logger = logging.CreateNewStdLoggerOrUseExistingLogger(logger)
+
+	logger.Print("Checking that all Nodes are running or completed...")
+
+	bv1C := k8sClient.BatchV1()
+	namespace := "openshift-monitoring"
+	name := "osd-cluster-ready"
+	jobs, err := bv1C.Jobs(namespace).List(ctx, metav1.ListOptions{})
```

Comment:
Sigh... It does. I just did it a dumb way because I too am not very familiar with client-go.

Reply:
Wait! On second thought, it *does* need to be a list. I need the resourceVersion field *of the list itself* in order to initiate the watch later. This allows me to detect the creation and deletion of jobs matching my criteria since I performed the list.

Reply:
Ah, I understand.

If you use a zero resourceVersion, do you even need the initial get/list?